src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -1678,7 +1707,7 @@ internal CalendarId[] CalendarIds
                     CalendarId[] calendars = new CalendarId[23];
                     Debug.Assert(_sWindowsName != null, "[CultureData.CalendarIds] Expected _sWindowsName to be populated by already");
 
-                    int count = GlobalizationMode.UseNls
+                    int count = ShouldUseUserOverrideNlsData
```

Comment:
>ShouldUseUserOverrideNlsData [](start = 32, length = 28)

this will be same too. getting the list from ICU and getting the default from NLS. later in this method we set the default. so I guess in this place we always read the list without caring about user overrides and then later when setting the default we do the user overrides check.